### PR TITLE
feat: add suggestion to parameter-properties rule

### LINKED ISSUE #11: 🚀 Feature: Add suggestions to parameter-properties rule
### Bug Report Checklist

- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-erasable-syntax-only/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Right now, the `erasable-syntax-only/parameter-properties` rule only reports a message on parameter properties. It doesn't provide any suggestions for how to fix them. Let's add one in to switch to traditional class properties.

### Additional Info

As a part of this work, the `docs/rules/parameter-properties.md` docs file should be added to as well. Right now it just has a straightforward codeblock. A PR fixing this issue should:

* Put a sub-heading on top of the existing example enum: `## Invalid Code`
* Add another code block under a `## Valid Code` with what the rule would suggest

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #11
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/eslint-plugin-erasable-syntax-only/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/eslint-plugin-erasable-syntax-only/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Adds a _Replace the parameter property with a class property._ suggestion to the `parameter-properties` rule. It moves the parameter's modifiers and name onto a class property declared right before the constructor, then assigns the parameter to it inside the constructor:

```ts
class Values {
	readonly value: number;
	constructor(value: number) {
		this.value = value;
	}
}
```

❎

### CODE REVIEW

File: src/rules/parameter-properties.ts [hunk 1]
```diff
@@ -1,12 +1,103 @@
...
+
+function getPropertyName(node: TSESTree.TSParameterProperty) {
+	const { parameter } = node;
+	const name =
+		parameter.type === AST_NODE_TYPES.AssignmentPattern
+			? parameter.left
+			: parameter;
+
+	// Class properties need a type annotation: unlike parameters, they can't
+	// infer one from a default value.
```
Comment (JoshuaKGoldberg) [inline src/rules/parameter-properties.ts]:
```suggestion
```

File: src/rules/parameter-properties.ts [hunk 2]
```diff
@@ -1,12 +1,103 @@
...
+			if (
+				!name ||
+				node.decorators.length ||
+				method.type !== AST_NODE_TYPES.MethodDefinition ||
+				body?.type !== AST_NODE_TYPES.BlockStatement
+			) {
+				return undefined;
+			}
+
+			// Assignments in a derived class' constructor must come after super().
```
Comment (JoshuaKGoldberg) [inline src/rules/parameter-properties.ts]:
```suggestion
```

**Comment by JoshuaKGoldberg** [comment]
Went back over this looking specifically for syntax edge cases the tests weren't covering. That turned up two real defects in the fix, both now fixed, plus 16 new test cases.

**Constructor overload signatures produced uncompilable code.** The property was inserted directly before the implementation, which separated it from its overload signature:

```ts
class Values {
	constructor(value: number);
	private value: number; // 💥 TS2390: Constructor implementation is missing.
	constructor(value: number) {
		this.value = value;
	}
}
```

The property now goes before the *first* constructor member of the class, so overloads stay adjacent to their implementation.

**A constructor body holding only comments got a stray whitespace line**, because emptiness was measured in statements alone. It now also checks for comments inside the body.

New cases, chosen to hit distinct syntax rather than to pad the count:

- `override readonly` modifiers, `protected constructor`, class expressions, generic-free multiline type annotations
- constructor overload signatures, `super()` preceded by other statements, `super()` only inside an `if` (no suggestion, since there's no single point that runs before `this` is used)
- comment-only constructor body, comments interleaved through the modifiers, sibling members and a trailing method around the constructor
- an arrow function default value (an `=` inside the parameter), a mix of typed and untyped parameter properties in one constructor
- indentation inference: one-line class, four-space indentation, and a class nested inside a function

To check the suggestions rather than just their shape, I extracted the output of every suggestion the test file produces and ran `tsc --strict --erasableSyntaxOnly` over all 20 of them. All compile. The only diagnostics left are expected: applying one suggestion in a two-parameter-property constructor leaves the other parameter property, which is still an `erasableSyntaxOnly` violation until its own suggestion is applied.

One behavior I verified rather than assumed, since it decides whether this transform is semantics-preserving: TypeScript's own emit under `useDefineForClassFields: true` declares the field first and assigns it in the constructor after `super()` - exactly the shape this suggests. Under the legacy `useDefineForClassFields: false`, TS instead assigns parameter properties *before* field initializers, so a class whose field initializer reads `this.value` would observe a different value. That mode is the opposite of what `erasableSyntaxOnly` users target, so I left it alone.

❎

File: src/rules/parameter-properties.ts [hunk 3]
```diff
@@ -1,12 +1,112 @@
...
+			}
+
+			const { superClass } = method.parent.parent;
+			const superCall = superClass ? getSuperCall(body) : undefined;
+
+			if (superClass && !superCall) {
+				return undefined;
+			}
+
+			// Overload signatures must stay adjacent to their implementation.
```
Comment (JoshuaKGoldberg) [inline src/rules/parameter-properties.ts]:
```suggestion
```
